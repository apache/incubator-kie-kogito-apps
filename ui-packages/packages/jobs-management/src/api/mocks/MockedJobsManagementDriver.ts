import { JobsManagementDriver } from '../JobsManagementDriver';

export const MockedJobsManagementDriver = jest.fn<JobsManagementDriver, []>(
  () => ({
    initialLoad: jest.fn(),
    applyFilter: jest.fn(),
    bulkCancel: jest.fn(),
    cancelJob: jest.fn(),
    rescheduleJob: jest.fn(),
    sortBy: jest.fn(),
    query: jest.fn()
  })
);
