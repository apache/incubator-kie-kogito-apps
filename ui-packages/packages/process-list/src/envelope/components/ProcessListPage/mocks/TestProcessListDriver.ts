/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { ProcessInstance } from '@kogito-apps/management-console-shared';
import {
  ProcessInstanceFilter,
  ProcessListDriver,
  SortBy
} from '../../../../api';

export default class TestProcessListDriver implements ProcessListDriver {
  private readonly processInstances: ProcessInstance[];
  private readonly childProcessInstances: ProcessInstance[];
  private offset: number = 0;
  private limit: number = 10;

  constructor(
    processInstances: ProcessInstance[],
    childProcessInstances: ProcessInstance[]
  ) {
    this.processInstances = processInstances;
    this.childProcessInstances = childProcessInstances;
  }
  /* eslint-disable  @typescript-eslint/no-unused-vars */
  private doSetState(processListFilter: ProcessInstanceFilter, sortBy: SortBy) {
    // do nothing
  }

  initialLoad(filter: ProcessInstanceFilter, sortBy: SortBy): Promise<void> {
    this.doSetState(filter, sortBy);
    return Promise.resolve();
  }

  openProcess(process: ProcessInstance): Promise<void> {
    return Promise.resolve();
  }
  applyFilter(filter: ProcessInstanceFilter): Promise<void> {
    // do nothing
    return Promise.resolve();
  }

  applySorting(sortBy: SortBy): Promise<void> {
    // do nothing
    return Promise.resolve(undefined);
  }

  query(offset: number, limit: number): Promise<ProcessInstance[]> {
    this.offset = offset;
    this.limit = limit;
    return this.doQuery(offset, this.getQueryLimit());
  }

  getChildProcessesQuery(
    rootProcessInstanceId: string
  ): Promise<ProcessInstance[]> {
    return Promise.resolve(this.childProcessInstances.slice(0, 10));
  }
  /* eslint-enable  @typescript-eslint/no-unused-vars */
  private doQuery(start: number, limit: number): Promise<ProcessInstance[]> {
    const queryLimit =
      limit > this.processInstances.length
        ? this.processInstances.length
        : limit;

    return Promise.resolve(this.processInstances.slice(start, queryLimit));
  }

  private getQueryLimit = (): number => {
    return this.offset + this.limit;
  };
}
