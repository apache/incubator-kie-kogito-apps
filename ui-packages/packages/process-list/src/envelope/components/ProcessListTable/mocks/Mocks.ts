/*
 * Copyright 2021 Red Hat, Inc. and/or its affiliates.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {
  ProcessInstance,
  ProcessInstanceState
} from '@kogito-apps/management-console-shared';
export const ProcessInstances: ProcessInstance[] = [
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  },
  {
    id: '538f9feb-5a14-4096-b791-2055b38da7c6',
    processId: 'travels',
    businessKey: 'Tra234',
    parentProcessInstanceId: null,
    parentProcessInstance: null,
    processName: 'travels',
    rootProcessInstanceId: null,
    roles: [],
    state: ProcessInstanceState.Error,
    addons: ['jobs-management', 'prometheus-monitoring', 'process-management'],
    start: new Date('2019-10-22T03:40:44.089Z'),
    error: {
      nodeDefinitionId: '__a1e139d5-4e77-48c9-84ae-34578e9817n',
      message: 'Something went wrong'
    },
    lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
    serviceUrl: 'http://localhost:4000',
    endpoint: 'http://localhost:4000',
    variables:
      '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-23T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-23T22:00:00Z[UTC]","city":"New York","country":"US","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"hotel":{"address":{"city":"New York","country":"US","street":"street","zipCode":"12345"},"bookingNumber":"XX-012345","name":"Perfect hotel","phone":"09876543"},"traveller":{"address":{"city":"Berlin","country":"Germany","street":"Bakers","zipCode":"100200"},"email":"cristiano@redhat.com","firstName":"Cristiano","lastName":"Nicolai","nationality":"German"}}',
    nodes: [],
    milestones: [],
    childProcessInstances: [
      {
        id: 'c54ca5b0-b975-46e2-a9a0-6a86bf7ac21eajabbcc',
        processId: 'travels',
        businessKey: 'TP444',
        parentProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        parentProcessInstance: null,
        rootProcessInstanceId: '538f9feb-5a14-4096-b791-2055b38da7c6',
        processName: 'FlightBooking test 2',
        roles: [],
        state: ProcessInstanceState.Active,
        serviceUrl: 'http://localhost:4000',
        lastUpdate: new Date('2019-10-22T03:40:44.089Z'),
        endpoint: 'http://localhost:4000',
        addons: ['process-management'],
        error: {
          nodeDefinitionId: 'a1e139d5-81c77-48c9-84ae-34578e90433n',
          message: 'Something went wrong'
        },
        start: new Date('2019-10-22T03:40:44.089Z'),
        end: new Date('2019-10-22T03:40:44.089Z'),
        variables:
          '{"flight":{"arrival":"2019-10-30T22:00:00Z[UTC]","departure":"2019-10-22T22:00:00Z[UTC]","flightNumber":"MX555"},"trip":{"begin":"2019-10-22T22:00:00Z[UTC]","city":"Berlin","country":"Germany","end":"2019-10-30T22:00:00Z[UTC]","visaRequired":false},"traveller":{"address":{"city":"Karkow","country":"Poland","street":"palna","zipCode":"200300"},"email":"rob@redhat.com","firstName":"Rob","lastName":"Rob","nationality":"Polish"}}',
        nodes: [],
        milestones: [],
        childProcessInstances: []
      }
    ]
  }
];
