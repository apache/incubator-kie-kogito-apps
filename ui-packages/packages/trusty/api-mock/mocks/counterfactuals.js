const counterfactuals = {
  executionId: 'executionId',
  counterfactualId: 'counterfactualId',
  goals: [],
  searchDomains: [],
  solutions: []
};

module.exports = counterfactuals;
