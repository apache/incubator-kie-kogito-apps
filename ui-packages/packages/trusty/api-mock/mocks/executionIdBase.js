module.exports = "b2b0ed8d-c1e2-46b5-3ac54ff4beae-";
