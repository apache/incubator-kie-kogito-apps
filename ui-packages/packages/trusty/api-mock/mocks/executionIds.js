module.exports = [
  'strings--4979-4e03-8692-0ec45cfca6ac',
  'unsuppor-ted--outc-omes-9437efcd449a',
  'unsuppor-ted--inpu-ts---a0741e0e7272',
  'unsuppor-ted--all--9512-9d6c78eb184b',
  'unsuppor-ted--all--stri-913ac955b90b',
  'ac6d2f5f-4eba-4557-9d78-22b1661a876a',
  '4fbeb765-1e8a-49ad-b833-25351ffc34e6',
  '3520c269-e28c-42ec-a930-27db76c0af37',
  'fe7ad865-0705-4807-a2a6-c98d362ab221',
  '3a563be3-42db-481d-ba01-25533bf4af83'
];
