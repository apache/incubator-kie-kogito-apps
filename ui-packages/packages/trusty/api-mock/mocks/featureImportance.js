const featureImportance = [];

const features = [
  "Monthly Tax Payment",
  "Monthly Insurance Payment",
  "Monthly HOA Payment",
  "Credit Score",
  "Down Payment",
  "Employment Income",
  "Other Income",
  "Assets",
  "Liabilities",
  "Lender Ratings",
  "Other Income 2",
  "Assets 2",
  "Liabilities 2",
  "Lender Ratings 2",
];

for (let i = features.length - 1; i > 0; i--) {
  const j = Math.floor(Math.random() * i);
  const temp = features[i];
  features[i] = features[j];
  features[j] = temp;
}

for (let i = 0; i < features.length - 1; i++) {
  const feature = {
    featureName: features[i],
    featureScore: Math.random() * (Math.random() > 0.5 ? 1 : -1),
  };
  featureImportance.push(feature);
}

exports.featureImportance = featureImportance;
