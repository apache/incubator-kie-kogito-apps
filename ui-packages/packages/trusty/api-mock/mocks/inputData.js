const inputData = [
    {
        "name": "Credit Score",
        "typeRef": "number",
        "value": 738,
        "components": []
    },
    {
        "name": "Down Payment",
        "typeRef": "number",
        "value": 70000,
        "components": []
    },
    {
        "name": "Property",
        "typeRef": "tProperty",
        "value": null,
        "components": [
            {
                "name": "Purchase Price",
                "typeRef": "number",
                "value": 34000,
                "components": []
            },
            {
                "name": "Monthly Tax Payment",
                "typeRef": "number",
                "value": 0.2,
                "components": []
            },
            {
                "name": "Monthly Insurance Payment",
                "typeRef": "number",
                "value": 0.15,
                "components": []
            },
            {
                "name": "Monthly HOA Payment",
                "typeRef": "number",
                "value": 0.12,
                "components": []
            },
            {
                "name": "Address",
                "typeRef": "tAddress",
                "value": null,
                "components": [
                    {
                        "name": "Street",
                        "typeRef": "string",
                        "value": "272 10th St.",
                        "components": []
                    },
                    {
                        "name": "Unit",
                        "typeRef": "string",
                        "value": "A",
                        "components": []
                    },
                    {
                        "name": "City",
                        "typeRef": "string",
                        "value": "Malibu",
                        "components": []
                    },
                    {
                        "name": "State",
                        "typeRef": "string",
                        "value": "CA",
                        "components": []
                    },
                    {
                        "name": "ZIP",
                        "typeRef": "string",
                        "value": "90903",
                        "components": []
                    }
                ],
            }
        ],
    },
    {
        "name": "Borrower",
        "typeRef": "tBorrower",
        "value": null,
        "components": [
            {
                "name": "Full Name",
                "typeRef": "string",
                "value": "Jim Osterberg",
                "components": []
            },
            {
                "name": "Tax ID",
                "typeRef": "string",
                "value": "11123322323",
                "components": []
            },
            {
                "name": "Employment Income",
                "typeRef": "number",
                "value": 99000,
                "components": []
            },
            {
                "name": "Other Income",
                "typeRef": "number",
                "value": 0,
                "components": []
            },
            {
                "name": "Assets",
                "typeRef": "tAssets",
                "value": null,
                "components": [
                    [
                        {
                            name: "Type",
                            typeRef: "string",
                            value: "C",
                            components: []
                        },
                        {
                            name: "Institution Account or Description",
                            typeRef: "string",
                            value: "Chase",
                            components: []
                        },
                        {
                            name: "Value",
                            typeRef: "number",
                            value: 45000,
                            components: []
                        }
                    ],
                    [
                        {
                            name: "Type",
                            typeRef: "string",
                            value: "Other Non-Liquid",
                            components: []
                        },
                        {
                            name: "Institution Account or Description",
                            typeRef: "string",
                            value: "Vanguard",
                            components: []
                        },
                        {
                            name: "Value",
                            typeRef: "number",
                            value: 33000,
                            components: []
                        }
                    ]
                ]
            }
        ],
    },
    {
        "name": "Liabilities",
        "typeRef": "tLiabilities",
        "value": null,
        "components": [
            [
                {
                    name: "Type",
                    value: "Credit Card",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Payee",
                    value: "Chase",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Monthly Payment",
                    value: 300,
                    typeRef: "number",
                    components: []
                },
                {
                    name: "Balance",
                    value: 0,
                    typeRef: "number",
                    components: []
                },
                {
                    name: "To be paid off",
                    value: "Yes",
                    typeRef: "string",
                    components: []
                }
            ],
            [
                {
                    name: "Type",
                    value: "Lease",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Payee",
                    value: "BMW Finance",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Monthly Payment",
                    value: 450,
                    typeRef: "number",
                    components: []
                },
                {
                    name: "Balance",
                    value: 0,
                    typeRef: "number",
                    components: []
                },
                {
                    name: "To be paid off",
                    value: "No",
                    typeRef: "string",
                    components: []
                }
            ]
        ]
    },
    {
        "name": "Lender Ratings",
        "typeRef": "tLenderRatings",
        "value": null,
        "components": [
            [
                {
                    name: "Lender Name",
                    value: "Gordon Cole",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Customer Rating",
                    value: 4.2,
                    typeRef: "number",
                    components: []
                }
            ],
            [
                {
                    name: "Lender Name",
                    value: "Dale Cooper",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Customer Rating",
                    value: 3.6,
                    typeRef: "number",
                    components: []
                }
            ],
            [
                {
                    name: "Lender Name",
                    value: "Chester Desmond",
                    typeRef: "string",
                    components: []
                },
                {
                    name: "Customer Rating",
                    value: 4.6,
                    typeRef: "number",
                    components: []
                }
            ]
        ]
    }
];

exports.inputs = inputData;