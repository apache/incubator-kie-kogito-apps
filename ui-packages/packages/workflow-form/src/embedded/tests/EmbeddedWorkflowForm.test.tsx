import { render } from '@testing-library/react';
import React from 'react';
import {
  EmbeddedWorkflowForm,
  EmbeddedWorkflowFormProps
} from '../EmbeddedWorkflowForm';
import { MockedWorkflowFormDriver } from './mocks/Mocks';

describe('EmbeddedWorkflowForm tests', () => {
  it('Snapshot', () => {
    const props: EmbeddedWorkflowFormProps = {
      driver: new MockedWorkflowFormDriver(),
      workflowDefinition: null,
      targetOrigin: 'origin'
    };

    const container = render(<EmbeddedWorkflowForm {...props} />).container;

    expect(container).toMatchSnapshot();

    const contentDiv = container.querySelector('div');

    expect(contentDiv).toBeTruthy();
  });
});
