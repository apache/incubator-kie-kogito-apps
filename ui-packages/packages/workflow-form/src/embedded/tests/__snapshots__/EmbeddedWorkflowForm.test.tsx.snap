// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbeddedWorkflowForm tests Snapshot 1`] = `
<div>
  <div />
</div>
`;
